sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -514,18 +514,27 @@ export class BatchingReceiverLite {
     };
 
     abortSignalCleanupFunction = checkAndRegisterWithAbortSignal((err) => {
+      if (receiver.drain) {
+        // If a drain is already in process and we cancel, the link state may be out of sync
+        // with remote. Reset the link so that we will have fresh start.
+        receiver.close();
+        return;
+      }
       reject(err);
```

Comment:
But would the promise be rejected if we return that early?

Reply:
the `return` was there before my change.. I only added code to close the link, so that the link will be re-opened.  After we drain, we expect a drained event then will resolve in the onDrain handler.  This check was for preventing re-entrancy I think

Reply:
hmm wait it was not there.

Reply:
I remember now: it cames with copy-n-paste of code above...removing it